test/test_autograd.py

```diff
@@ -2536,6 +2536,20 @@ def test_profiler(self):
             self.assertEqual(info.name, expected_name)
             last_end = info.cpu_interval.end
 
+    def test_record_function_callbacks(self):
+        x = torch.randn(10, 10)
+        with profile() as p:
+            rf = torch.autograd._RecordFunction()
+            torch.autograd._run_before_callbacks(rf, "foo")
+            y = x * 2 + 4
+            # ensure that we run destructor for RecordFunction, which invokes
+            # end callbacks
+            del rf
+            gc.collect()
+        function_events = p.function_events
+        foo_event = [event for event in function_events if "foo" in event.name][0]
+        self.assertEqual(foo_event.count, 1)
+
```

Comment:
What is this unit test testing that is not already covered by the `test_profiler_*` tests above?

Reply:
Nothing of significance, though I still decided to add this test because the `test_profiler_` tests live in the RPC module. If someone is modifying the autograd profiler outside of the RPC context they will likely run `test_autograd.py` during development, and catch issues quicker. Should we take it out?